.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,5 @@
-Make sure you have checked _all_ steps below.
-
-### Jira
-
-- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
-  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
-  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
-  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
```

Comment:
I think this is a useful information to keep so that users are aware of this requirement.

Reply:
I added information (not a tick-mark) about AIP. That's useful. The rest is just redundant with "Commit message contains [\[AIRFLOW-XXXX\]](https://issues.apache.org/jira/browse/AIRFLOW-XXXX) or `[AIRFLOW-XXXX]` for document-only changes" I think.

One of the goals of that change was to fit every description in one line of github comment and to fit everything in the default size of the PR description box. It's pretty much guaranteed it will not be read if it's longer/multi-line and we can always explain it during review process if someone did not understand it (plus we have very nice description in CONTRIBUTING.md that we can refer people to). I actually added a line with link to contributing now.